esp32/mods/modbt.c

```diff
@@ -517,29 +487,42 @@ static bool pin_changed(uint32_t new_pin)
      return ret;
 }
 
-static void remove_all_bonded_devices(void)
+static void remove_bonded_devices(void)
 {
     int dev_num = esp_ble_get_bond_device_num();
 
     esp_ble_bond_dev_t *dev_list = heap_caps_malloc(sizeof(esp_ble_bond_dev_t) * dev_num, MALLOC_CAP_INTERNAL | MALLOC_CAP_8BIT);
     esp_ble_get_bond_device_list(&dev_num, dev_list);
     for (int i = 0; i < dev_num; i++) {
+        if (bt_obj.gatts_conn_id >= 0 && memcmp((void *) dev_list[i].bd_addr, (void *) bt_obj.client_bda, sizeof(esp_bd_addr_t)) == 0) {
+            continue;
+        }
```

Comment:
If a see correctly, the aim of this part is to not remove bond with device which is currently connected.

I tested this part, and bond won't be removed between LoPy4 and connected peripheral, but as you disconnect, can't be reconnect and rebond to LoPy4 anymore, I could not seen its advertisement anymore.

Reply:
This was tested and confirmed to work as expected on a LoPy4, FiPy and GPy, and we are seeing advertisements in the raw traffic. The sending of advertisements is not handled by this part of the code, so I'm not sure what causes the issue in your test but it seems like it wouldn't be related to the changes in these lines.

Reply:
Yes, I have also tested, and I am sure that the problem is not coming from your changes, It is also an issue without it. I have tested with `LoPy4` and `nRF Connect` on my Android, and it both cases, I experience that:

1. Connect to LoPY4
2. Modify secured characteristics
3. Bonding devices
4. Set new pin
5. Disconnect (with your change) `or` Automatically disconnected (withhout the change)
6. Advertisement disappears (with manual `bluetooth.advertise(True)` it can be seen again)

So I recommended to add `esp_ble_gap_start_advertising(&bt_adv_params);` at the end of `remove_bonded_devices()` function. 

1. 

1.